lib/Botkit.d.ts

```diff
@@ -0,0 +1,482 @@
+    appId: string;
+    appPassword: string;
+  }
+  interface Channel {
+    id: string;
+  }
+  interface CiscoSparkBot extends Bot<CiscoSparkSpawnConfiguration, CiscoSparkMessage> {
+    retrieveFile(url: string, cb: (err: Error, body: any) => void): void;
+    retrieveFileInfo(url: string, cb: (err: Error, obj: any) => void): void;
+    startPrivateConversation(message: CiscoSparkMessage, cb: (err: Error, convo: Conversation<CiscoSparkMessage>) => void): void;
+    startPrivateConversationWithActor(message: CiscoSparkMessage, cb: (err: Error, convo: Conversation<CiscoSparkMessage>) => void): void;
+    startPrivateConversationWithPersonId(personId: string, cb: (err: Error, convo: Conversation<CiscoSparkMessage>) => void): void;
+  }
+  interface CiscoSparkConfiguration extends Configuration {
+    ciscospark_access_token: string;
+    limit_to_domain?: string | string[];
+    limit_to_org?: string;
+    public_address: string;
+    secret?: string;
+    webhook_name?: string;
+  }
+  interface CiscoSparkController extends Controller<CiscoSparkSpawnConfiguration, CiscoSparkMessage, CiscoSparkBot> {
+    createWebhookEndpoints(webserver: any, bot: CiscoSparkBot, cb?: () => void): this;
+  }
+  interface CiscoSparkMessage extends Message {
+    actorId?: string;
+    data?: {
+      personDisplayName: string;
+    };
+    files?: any[];
+    markdown?: string;
+    original_message?: CiscoSparkMessage;
+  }
+  interface CiscoSparkSpawnConfiguration {
+  }
+  interface Configuration {
+    debug?: boolean;
+    hostname?: string;
+    json_file_store?: string;
+    log?: boolean;
+    logger?: { log: Function; };
+    storage?: {
+      users: Storage<User>;
+      channels: Storage<Channel>;
+      teams: Storage<Team>;
+    };
+    studio_token?: string;
+  }
+  interface ConsoleBot extends Bot<ConsoleSpawnConfiguration, ConsoleMessage> {
+  }
+  interface ConsoleConfiguration extends Configuration {
+  }
+  interface ConsoleController extends Controller<ConsoleSpawnConfiguration, ConsoleMessage, ConsoleBot> {
+  }
+  interface ConsoleMessage extends Message {
+  }
+  interface ConsoleSpawnConfiguration {
+  }
+  interface Controller<S, M extends Message, B extends Bot<S, M>> {
+    readonly storage: {
+      users: Storage<User>;
+      channels: Storage<Channel>;
+      teams: Storage<Team>;
+    };
+    readonly studio: Studio<S, M, B>;
+    readonly log: {
+      (...params: any[]): void;
+    }
+    hears(keywords: string | string[] | RegExp | RegExp[], events: string | string[], cb: HearsCallback<S, M, B>): this;
+    hears(keywords: string | string[] | RegExp | RegExp[], events: string | string[], middleware_or_cb: HearsFunction<M>, cb: HearsCallback<S, M, B>): this;
+    on(event: string, cb: HearsCallback<S, M, B>): this;
+    setupWebserver(port: number | string, cb: (err: Error, webserver: any) => void): this;
+    spawn(config?: S, cb?: (worker: B) => void): B;
+    startTicking(): void;
```

Comment:
middleware property is not declared.
https://github.com/howdyai/botkit/blob/aeb1b025f1fd8f69879ebe33096ad3bd3eb63456/lib/CoreBot.js#L42